src/app/features/add-course/add-course-window/add-course-window.component.ts

```diff
@@ -48,20 +50,61 @@ export class AddCourseWindowComponent implements OnInit {
     this.courseForm.valueChanges.subscribe(value => {
       this.formValue = value;
     });
+
+    this.getCourse();
+    this.checkForm();
+    this.cdr.detectChanges();
   }
 
   get f() { return this.courseForm.controls; }
 
+  getCourse() {
+    let id = +this.activatedRoute.snapshot.paramMap.get('id');
+    if(id) {
+      this.course = this.courseService.getItem(id);
+      this.title = this.course.title;
```

Comment:
Not mistake, but improvement
Too many conversions. You could just import one Object course and send it to server (without creating many properties)